.buildkite/filebeat/filebeat-pipeline.yml

```diff
@@ -1,5 +1,157 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
 
+env:
+  SETUP_GVM_VERSION: "v0.5.1"
+  IMAGE_UBUNTU_X86_64: "family/core-ubuntu-2204"
+  IMAGE_UBUNTU_ARM_64: "core-ubuntu-2004-aarch64"
+  IMAGE_WIN_2016: "family/core-windows-2016"
+  IMAGE_WIN_2019: "family/core-windows-2019"
+  IMAGE_WIN_2022: "family/core-windows-2022"
+  DEBIAN_FRONTEND: "noninteractive"
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Filebeat Mandatory Testing"
+    key: "mandatory-tests"
+    if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "filebeat" || build.env("BUILDKITE_PULL_REQUEST") != "false"
+
+    steps:
+      - label: ":ubuntu: Unit Tests"
+        command:
+          - "sudo apt-get update"
+          - "sudo apt-get install -y libsystemd-dev"
+          - "sudo apt install -y python3-pip"
+          - "sudo apt-get install -y python3-venv"
+          - ".buildkite/filebeat/scripts/unit-tests.sh"
+        notify:
+          - github_commit_status:
+              context: "Filebeat/Mandatory: Unit Tests"
```

Comment:
I'm not totally onboarded for using `mandatory`; IIUC, the GitHub branch protection can enforce what GitHub checks should be mandatory instead; what do you think?

Reply:
Thanks for idea, confirmed with Julien to implement it the end of beats migration.